internal/extproc/server.go

```diff
@@ -125,6 +162,16 @@ func (s *Server[P]) process(p P, stream extprocv3.ExternalProcessor_ProcessServe
 			return status.Errorf(codes.Unknown, "cannot receive stream request: %v", err)
 		}
 
+		// If we're processing the request headers, read the :path header to instantiate the
+		// right processor.
+		if headers := req.GetRequestHeaders().GetHeaders(); headers != nil {
+			p, err = s.processorForPath(headersToMap(headers))
```

Comment:
I think this will create a new `p` each time the grpc stream receives a new message? maybe this block should be conditioned on `if p != nil { ... }` ?

Reply:
ah `headers := req.GetRequestHeaders().GetHeaders(); headers != nil` implicitly does that - maybe I would like a comment at least on this

Reply:
The `req.GetRequestHeaders()` will only be non-nil when it is a RequestHeaders message, so I think this will be executed only once?
```go
func (x *ProcessingRequest) GetRequestHeaders() *HttpHeaders {
	if x, ok := x.GetRequest().(*ProcessingRequest_RequestHeaders); ok {
		return x.RequestHeaders
	}
	return nil
}
```

Reply:
I'll add a comment :)

Reply:
Done